app/v3/connectors/RetrieveForeignPropertyBsasConnector.scala

```diff
@@ -18,30 +18,44 @@ package v3.connectors
 
 import config.AppConfig
 
-import javax.inject.{Inject, Singleton}
+import javax.inject.{ Inject, Singleton }
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.http.HttpClient
-import v3.connectors.DownstreamUri.DesUri
+import v3.connectors.DownstreamUri.{ IfsUri, TaxYearSpecificIfsUri }
 import v3.models.request.retrieveBsas.foreignProperty.RetrieveForeignPropertyBsasRequestData
 import v3.models.response.retrieveBsas.foreignProperty.RetrieveForeignPropertyBsasResponse
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.{ ExecutionContext, Future }
 import v3.connectors.httpparsers.StandardDesHttpParser._
+import v3.models.domain.TaxYear
 
 @Singleton
-class RetrieveForeignPropertyBsasConnector @Inject()(val http: HttpClient,
-                                                     val appConfig: AppConfig) extends BaseDownstreamConnector {
+class RetrieveForeignPropertyBsasConnector @Inject()(val http: HttpClient, val appConfig: AppConfig) extends BaseDownstreamConnector {
 
   def retrieveForeignPropertyBsas(request: RetrieveForeignPropertyBsasRequestData)(
-    implicit hc: HeaderCarrier,
-    ec: ExecutionContext,
-    correlationId: String): Future[DownstreamOutcome[RetrieveForeignPropertyBsasResponse]] = {
+      implicit hc: HeaderCarrier,
+      ec: ExecutionContext,
+      correlationId: String): Future[DownstreamOutcome[RetrieveForeignPropertyBsasResponse]] = {
+
+    import request._
+
+    val taxYearVal = taxYear match {
```

Comment:
Maybe match on the tax year to give the url, then you won't have to include a tax year that will never be used. Something like (bearing in mind the taxYear will be a `TaxYear`):
```scala
val url = taxYear match {
  case Some(ty) if ty.useTaxYearSpecificApi => TaxYearSpecificIfsUri[...]("...")
  case _ => IfsUri[...]("..."))
}
```